apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,200 @@
+    const localDeparture = this.parseDate(routeItem?.departure?.localTime);
+    const utcArrival = this.parseDate(routeItem?.arrival?.utcTime);
+    const utcDeparture = this.parseDate(routeItem?.departure?.utcTime);
+    const duration =
+      routeItem &&
+      `${DateFNS.differenceInHours(
+        utcArrival,
+        utcDeparture,
+      )}h ${DateFNS.differenceInMinutes(utcArrival, utcDeparture) % 60}m`;
+
+    return {
+      arrival: (routeItem && routeItem.arrival?.city) ?? null,
+      arrivalTime:
+        (localArrival && DateFNS.format(localArrival, timeSimpleFormat)) ??
+        null,
+      carrier: { code: (routeItem && routeItem.airline) ?? null, name: null }, // @TODO: get name of the airline
+      tripDate:
+        (localDeparture && DateFNS.format(localDeparture, dateFormat)) ?? null,
+      departure: (routeItem && routeItem.departure?.city) ?? null,
+      departureTime:
+        (localDeparture && DateFNS.format(localDeparture, timeSimpleFormat)) ??
+        null,
+      duration: duration ?? null,
+      id: (routeItem && routeItem.id) ?? null,
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
+    const isOneWay = routes != null && routes.length < 2;
+    if (!route || isOneWay) {
+      return null;
+    }
+    return route.findIndex(
+      routeItem => routeItem?.departure?.cityCode === routes?.[1]?.[0],
+    );
+  };
+
+  // @TODO: make ConnectionCard more universal (not only for forth and back routes)
+  getSanitizedRoute = (): Array<TripSegmentWithId> => {
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    sanitizeRoute: Array<TripSegmentWithId>,
+    demandedSector: number,
+    sectorBorderIndex: ?number,
+  ): Array<TripSegmentWithId> => {
+    if (sectorBorderIndex != null) {
+      if (demandedSector === 0) {
+        return sanitizeRoute.slice(0, sectorBorderIndex);
+      }
+      return sanitizeRoute.slice(sectorBorderIndex);
+    }
+    return sanitizeRoute;
+  };
+
+  getDurationBetweenSectors = (route: ?Route, sectorBorderIndex: ?number) => {
+    if (route && sectorBorderIndex != null && sectorBorderIndex > 0) {
+      const stopBeforeSector = route[sectorBorderIndex - 1];
+      const stopAfterSector = route[sectorBorderIndex];
+      const utcDeparture = this.parseDate(stopAfterSector?.departure?.utcTime);
+      const utcArrival = this.parseDate(stopBeforeSector?.arrival?.utcTime);
+      return DateFNS.distanceInWords(utcDeparture, utcArrival);
+    }
+    return null;
+  };
+
+  render() {
+    const { data } = this.props;
+    const sanitizedRoute = this.getSanitizedRoute();
+    const sectorBorderIndex = this.getSectorBorderIndex();
+    const hasSectorBorder = sectorBorderIndex != null && sectorBorderIndex > -1;
+    if (data == null) {
```

Comment:
Shouldn't be better if this condition will be imidiatly after `const { data } = this.props`?

Reply:
OK, done